tools/build-tools/src/common/utils.ts

```diff
@@ -122,8 +122,8 @@ export function readJsonSync(filename: string) {
     return JSON.parse(content);
 }
 
-export async function lookUpDir(dir: string, callback: (currentDir: string) => boolean | Promise<boolean>) {
-    let curr = dir;
+export async function lookUpDirAsync(dir: string, callback: (currentDir: string) => boolean | Promise<boolean>) {
```

Comment:
>an | Promise<bo [](start = 89, length = 15)

using a union between  boolean | Promise<boolean> doesn't gain much, and makes the api harder to use, you should just do promise<boolean>

Reply:
especially now that we have a sync version

---
In reply to: [541326276](https://github.com/microsoft/FluidFramework/pull/4577#discussion_r541326276) [](ancestors = 541326276)

Reply:
You are right, I cleaned it up.